src/io.c

```diff
@@ -368,26 +368,26 @@ bcd_io_listener_fd(const struct bcd_io_listener *l)
 struct bcd_io_listener *
 bcd_io_listener_unix(const char *path, int backlog, bcd_error_t *error)
 {
-	struct bcd_io_listener *listener = malloc(sizeof *listener);
+	struct bcd_io_listener *listener = calloc(1, sizeof *listener);
 	struct sockaddr_un un;
 
 	if (listener == NULL)
 		return NULL;
 
 	if (*path != '/') {
 		bcd_error_set(error, 0, "listener requires full path");
-		return NULL;
+		goto error;
```

Comment:
We could move this check and the one below it before we allocate `listener`, so we don't need to free it immediately afterwards in the `error` section.